### DISCUSSION TITLE
Plasticity Hardening Model

### DISCUSSION THREAD

**Discussion by sSajjad90** [idaholab/moose#23621] [Q&A Modules: Solid mechanics] [answered]
Hello,
I'm working on precipitation hardening of aluminum alloys and have already written source code to assess microstructure development and yield stress. I'm now attempting to adapt the `IsotropicPlasticityStressUpdate.C` and `IsotropicPowerLawHardeningStressUpdate.C` source codes to utilize a new hardening model in plastic deformation based on the equations below.

![comment](https://user-images.githubusercontent.com/78385504/223219324-9334d455-3a61-4413-a4a0-9cb6d67896d9.png)

Do you know how I may update the `IsotropicPlasticityStressUpdate.C` and `IsotropicPowerLawHardeningStressUpdate.C` source codes or build a new one to adopt this hardening model?

Sincerely,
Sajjad

**Comment by hugary1995** [comment, marked as answer]
You should write a new material for this. `IsotropicPowerLawHardeningStressUpdate` is a decent example.
